### PR TITLE
feat: attempt to preserve stack alignment proof in AxEffect update

### PR DESCRIPTION
**Pull request by alexkeizer**
### Description:

Stacked on #207.

When an write to memory or a write to a register that is not SP is made, we update the proof of stack alignment. When a write to SP is made, we create a new mvar with the proof obligation that the new value is aligned, and store that mvar in a new `sideConditions` field. 

### Testing:

What tests have been run? Did `make all` succeed for your changes? Was
conformance testing successful on an Aarch64 machine? Yes

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Tactics/Sym/AxEffects.lean [hunk 1]
```diff
@@ -78,6 +78,10 @@ structure AxEffects where
   However, if SP is written to, no effort is made to prove alignment of the
   new value; the field will be set to `none` -/
   stackAlignmentProof? : Option Expr
+
+  /-- `sideContitions` are proof obligations that come up during effect
```
Comment (shigoel) [inline Tactics/Sym/AxEffects.lean:82]:
```suggestion
  /-- `sideConditions` are proof obligations that come up during effect
```

Comment (shigoel) [reply Tactics/Sym/AxEffects.lean:82]: These side conditions only stem from the SP alignment proof. That deserves a mention here.

File: Arm/Insts/Common.lean [hunk 2]
```diff
@@ -219,20 +219,32 @@ def CheckSPAlignment (s : ArmState) : Prop :=
 instance : Decidable (CheckSPAlignment s) := by unfold CheckSPAlignment; infer_instance
 
 @[state_simp_rules]
-theorem CheckSPAligment_of_w_different (h : StateField.GPR 31#5 ≠ fld) :
+theorem CheckSPAligment_w_different_eq (h : StateField.GPR 31#5 ≠ fld) :
   CheckSPAlignment (w fld v s) = CheckSPAlignment s := by
   simp_all only [CheckSPAlignment, state_simp_rules, minimal_theory, bitvec_rules]
 
+theorem CheckSPAligment_w_of_ne_sp_of (h : StateField.GPR 31#5 ≠ fld) :
```
Comment (shigoel) [inline Arm/Insts/Common.lean:226]: At first glance, theorems like `CheckAlignment*_sp_of` seem unnecessary to have, esp. in the face of those like `CheckSPAligment_*_eq`. Mention that they are useful to construct a non-effects proof for SP and point to `AxEffects`.

Comment (alexkeizer) [reply Arm/Insts/Common.lean]: I've added a comment

File: Tactics/Sym/AxEffects.lean (hunk 1 above)
Comment (alexkeizer) [reply Tactics/Sym/AxEffects.lean:82]: I've expanded the docstring

**Comment by alexkeizer** [comment]
@shigoel rebased on main and addressed comments!

**Pull request merged**